pkg/user/signer.go

```diff
@@ -410,3 +404,17 @@ func QueryAccount(ctx context.Context, conn *grpc.ClientConn, encCfg encoding.Co
 	accNum, seqNum = acc.GetAccountNumber(), acc.GetSequence()
 	return accNum, seqNum, nil
 }
+
+func (s *Signer) getSignatureV2(sequence uint64, signature []byte) signing.SignatureV2 {
+	sigV2 := signing.SignatureV2{
+		Data: &signing.SingleSignatureData{
+			SignMode:  signing.SignMode_SIGN_MODE_DIRECT,
+			Signature: signature,
+		},
+		Sequence: sequence,
+	}
+	if sequence == 0 {
+		sigV2.PubKey = s.pk
```

Comment:
It seems like the PubKey needs to be set for every message

Reply:
[Question] "needs" or "does not need"?

Reply:
Also, wondering whether this change aligns with the PR description or is an independent additional fix?

Reply:
> [Question] "needs" or "does not need"?

In the past it always needed the pubkey because of some bug in the SDK but I've since fixed it now so it no longer needs the pubKey

> Also, wondering whether this change aligns with the PR description or is an independent additional fix?

Yeah I could extract it out

Reply:
> Yeah I could extract it out

Good idea 👍